test/Unit/Durable/DurableTimerTests.cs

```diff
@@ -0,0 +1,260 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Test.Durable
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Threading;
+    using Microsoft.Azure.Functions.PowerShellWorker.Durable;
+    using Moq;
+    using Xunit;
+
+    public class DurableTimerTests
+    {
+        private const DateTimeKind utc = DateTimeKind.Utc;
+        private DateTime time = new DateTime(2020, 1, 1, 0, 0, 0, 0, utc);
+        private int longIntervalSeconds = 5;
+        private int shortIntervalSeconds = 3;
+        private OrchestrationInvoker _orchestrationInvoker = new OrchestrationInvoker();
+        private OrchestrationBindingInfo _orchestrationBindingInfo = new OrchestrationBindingInfo(
+                                                                            "ContextParameterName",
+                                                                            new OrchestrationContext());
+        private Mock<IPowerShellServices> _mockPowerShellServices = new Mock<IPowerShellServices>();
+        private DurableTimer _durableTimer = new DurableTimer();
+        private int _nextEventId = 1;
+
+        // Verifies that CreateTimer waits for the time elapsed 
+        [Theory]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        public void CreateTimerOrContinue_WaitsUntilTimerFires(bool timerScheduledAndFired, bool expectedWaitForStop)
+        {
+            int durationSeconds = longIntervalSeconds;
+            DateTime startTime = time;
```

Comment:
If it never changes, I suggest you just use the field directly, no need to create a copy.